lib/src/repo.rs

```diff
@@ -89,6 +89,13 @@ impl<'a> RepoRef<'a> {
             RepoRef::Mutable(repo) => repo.view(),
         }
     }
+
+    pub fn shortest_unique_id_prefix_len(&self, target_id_bytes: &[u8]) -> usize {
```

Comment:
@martinvonz do you have any thoughts on where this kind of `index + view` functions should be hosted?

I'm not going to add persistent storage, but both `revset::resolve_change_id()` and `templater::DivergentProperty` can leverage the current in-memory change_id_index.

Reply:
Good question. This seems good for now. Maybe it'll be clearer where they should go later.